### ISSUE TITLE
Readiness probe port in PR 1825. Is 8008 correct?

### ISSUE THREAD

**Issue by beneso** [zalando/postgres-operator#1911]
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** registry.opensource.zalan.do/acid/postgres-operator:v1.8.1
- **Where do you run it - cloud or metal?** Kubernetes Bare Metal K8s
- **Are you running Postgres Operator in production?** Not yet, POC for Grafana in progress
- **Type of issue?** Bug report/question

Hi, in [PR 1825](https://github.com/zalando/postgres-operator/pull/1825/files#diff-e9c5b05262207b16b959659256e773d9158a1710b89d1b73eca6453888563130R1067) readinessProbe is set to be generated for port 8008. That way the container is never ready, HTTP 503. Is this intended or should the port be 8080, where we do get HTTP 200 for path `/readiness`? More in [this review](https://github.com/zalando/postgres-operator/pull/1825#pullrequestreview-990700800).

Thanks for having a look!
Ondrej

**Comment by beneso** [comment]
A follow-up: if the readiness probe is manually edited from 8008 to 8080 in the generated stateful set and pods are restarted, they do pass into ready state and the desired number of replicas is created.
CRD unfortunately does not allow configuring this part of the resulting stateful set.

~~~
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /readiness
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 6
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 5

~~~

**Comment by k402xxxcenxxx** [comment]
Hi,
I'm also wandering if these is correct?
the rendered container spec in yaml will present as:
```yaml
...
  containers:
    - name: postgres
      image: registry.opensource.zalan.do/acid/spilo-14:2.1-p5
      ports:
        - containerPort: 8008
          protocol: TCP
        - containerPort: 5432
          protocol: TCP
        - containerPort: 8008
          protocol: TCP
...
```
the second 8008 is duplicated

**Comment by k402xxxcenxxx** [comment]
> Hi, I'm also wondering if this is correct? the rendered container spec in yaml will present as:
> 
> ```yaml
> ...
>   containers:
>     - name: postgres
>       image: registry.opensource.zalan.do/acid/spilo-14:2.1-p5
>       ports:
>         - containerPort: 8008
>           protocol: TCP
>         - containerPort: 5432
>           protocol: TCP
>         - containerPort: 8008
>           protocol: TCP
> ...
> ```
> 
> the second 8008 is duplicated

Oh I saw this should be fixed by the MR https://github.com/zalando/postgres-operator/pull/1864
And update to v1.8.1 to get this bug fix
this card should be closed

**Comment by FxKu** [comment]
@k402xxxcenxxx the port fix is included in 1.8.1

**Comment by k402xxxcenxxx** [comment]
> @k402xxxcenxxx the port fix is included in 1.8.1

Yes! I notice that.
@beneso I suggest to close this issue.

**Comment by beneso** [comment]
@k402xxxcenxxx agree.
Sorry guys, I missed the correction. Closing.
Thanks!
